install/helm/agones/templates/service-monitor.yaml

```diff
@@ -19,15 +19,23 @@ metadata:
   name: agones-controller-monitor
   namespace: {{ .Release.Namespace }}
   labels:
+{{- if $featureGates.SplitControllerAndExtensions }}
```

Comment:
Do you need to add `{{- $featureGates := include "agones.featureGates" . | fromYaml }}` at the top (like in https://github.com/googleforgames/agones/blob/main/install/helm/agones/templates/controller.yaml) for this to evaluate? Did you test trying it with the feature gate on & off and verify the labels showed up as expected?

Reply:
Yup, added. 
With feature on:
<img width="1224" alt="Screenshot 2023-06-05 at 2 28 37 PM" src="https://github.com/googleforgames/agones/assets/17622084/024bbed3-bbb6-4af2-94a4-117074bc30ee">
With feature off:
<img width="1224" alt="Screenshot 2023-06-05 at 2 30 33 PM" src="https://github.com/googleforgames/agones/assets/17622084/be927ebd-64b6-4870-805c-d5afa392229a">